examples/get-cdn-link.js

```diff
@@ -0,0 +1,46 @@
+// To run this script, use `node get-cdn-link.js`
+//
+// Uncomment this line to run on an insecure local environment, if you
+// like:
+// process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+const permSdk = require('@permanentorg/node-sdk');
+
+// You can pull the "permSession" and "permMFA" cookie values from
+// devtools on any request.  Include
+//
+// 'baseUrl': 'https://local.permanent.org/api'
+//
+// or similar if you want to run this locally or on another
+// environment (by default it will run against prod)
+
+const permanent = new permSdk.Permanent({
+    'sessionToken': 'permSession_COOKIE',
+    'mfaToken': 'permMFA_COOKIE',
+});
+
+run();
+
+function randomElement(array) {
+    return array[Math.floor(Math.random() * array.length)];
+}
+
+async function run() {
+    await permanent.init();
+    console.log('session started');
+
+    const privateRoot = await permanent.folder.getMyFilesFolder();
+    const randomChild = randomElement(privateRoot.ChildItemVOs);
+
+    // doing this specifically for a record
+    if (randomChild && randomChild.uploadFileName) {
```

Comment:
To some degree I realize that this design is hopefully going to be completely redone anyways, so we shouldn't be too nitpicky about any of these smaller changes, but can we have a publicly exposed helper function that's something like `permanent.record.isRecord` which specifically tests for the existence of `uploadFileName`? I don't think the user should have to do that here.

To avoid the issue that we ran into earlier this morning, I'd also suggest initially filtering out all the non-folder items to begin with so we can have a more useful error message and a folder isn't randomly selected forcing us to restart the script, maybe something like this:
```javascript
const childRecords = privateRoot.ChildItemVOs.filter((item) => permanent.record.isRecord(item)); // or whatever an isRecord helper is called
if (childRecords.length === 0) {
  console.log("No records in My Files folder");
} else {
  const randomChild = randomElement(childRecords);
  // ... insert everything else here ...
}

console.log('Done!');
```

Reply:
Given that you've already written `isItemARecord` I'm inclined to use it again.  I see it's protected - is that for a bigger reason that I don't know or because when you initially used it you only wanted it in the item resource?

Reply:
(Also, sorry, I skipped over the "yes this is a much better/more useful way to present this example!"  And thank you!)

Reply:
I believe IsItemARecord is only protected because I was only using it internally. Feel free to make it public!